dvc/repo/ls_url.py

```diff
@@ -0,0 +1,35 @@
+from dvc.exceptions import URLMissingError
+from dvc.fs import get_cloud_fs
+
+
+def parse_external_url(url):
```

Comment:
```suggestion
def _parse_external_url(url):
```

Reply:
I think this function could be used elsewhere (e.g. in `Repo.get_url()`) and should probably be unit-tested, so it feels public to me. For the same reasons, it should probably move to some common utility module.